brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java

```diff
@@ -65,11 +67,74 @@ public BrageEntryEventConsumer() {
     public Publication handleRequest(S3Event s3Event, Context context) {
         return attempt(() -> parseBrageRecord(s3Event))
                    .map(publication -> pushAssociatedFilesToPersistedStorage(publication, s3Event))
+                   .map(publication -> hasExistingCristinIdentifier(publication)
+                                           ? attemptToUpdateExistingPublication(publication, s3Event)
+                                           : createNewPublication(publication, s3Event))
+                   .orElseThrow(fail -> handleSavingError(fail, s3Event));
+    }
+
+    private static Publication injectAssociatedArtifacts(Publication publication, Publication existingPublication) {
```

Comment:
No checks if the associated artifacts already exists?
Also I believe there is a specific task for this.

Reply:
next task